### PR TITLE
feat(seve): add cell type filter to share url

### PR DESCRIPTION
**Pull request by seve**
#5218

Adds cell type filter to share url and it's analytics events

Also fixes ordering issue with loading genes from share url

### CODE REVIEW

File: frontend/src/views/WheresMyGeneV2/components/GeneSearchBar/components/ShareButton/index.tsx [hunk 1]
```diff
@@ -20,13 +20,51 @@ import {
...
+
+  const {
+    data: rawCellTypesByTissueName,
+    isLoading: isLoadingCellTypesByTissueName,
+  } = useCellTypesByTissueName(2);
+
+  const [cellTypesByTissueName, setCellTypesByTissueName] =
+    useState<CellTypeByTissueName>(EMPTY_OBJECT);
+
+  // This is needed
```
Comment (tihuan) [inline frontend/src/views/WheresMyGeneV2/components/GeneSearchBar/components/ShareButton/index.tsx:52]: can we add why this is needed? Thanks!

File: frontend/src/views/WheresMyGeneV2/components/GeneSearchBar/components/ShareButton/index.tsx [hunk 2]
```diff
@@ -40,8 +78,13 @@ export default function ShareButton(): JSX.Element {
       filters: selectedFilters,
       organism: selectedOrganismId,
       genes: selectedGenes,
+      cellTypes: filteredCellTypes,
     });
 
+    const filteredCellTypeIDs = filteredCellTypes.map(
```
Comment (tihuan) [inline frontend/src/views/WheresMyGeneV2/components/GeneSearchBar/components/ShareButton/index.tsx:84]: To keep the code DRY, maybe worth extracting a function to get `filteredCellTypeIDs` ? I see similar logic here and below 🙏 Thank you!

**Review by tihuan — APPROVED** [review]
Thanks for the PR, @seve ! Just some minor comments otherwise LGTM 🚀 👏

**Review by atarashansky — APPROVED** [review]

File: frontend/src/views/WheresMyGeneV2/components/HeatMap/index.tsx [hunk 3]
```diff
@@ -286,17 +286,19 @@ export default memo(function HeatMap({
...
+  };
+
+  // update displayedCellTypes and expandedTissues
+  useEffect(() => {
+    if (filteredCellTypes.length === 0) {
       setDisplayedCellTypes(initialDisplayedCellTypes);
       setExpandedTissues(new Set<string>());
-      setFilteredCellTypes(newFilteredCellTypes);
       return;
     }
```
Comment (seve) [inline frontend/src/views/WheresMyGeneV2/components/HeatMap/index.tsx:303]: Erroneous

Comment (seve) [inline frontend/src/views/WheresMyGeneV2/components/HeatMap/index.tsx:303]: Erroneous

**Pull request merged**